core/src/main/java/com/linecorp/armeria/internal/http/Http1ObjectEncoder.java

```diff
@@ -344,9 +343,23 @@ private ChannelFuture doWriteSplitData(
         }
 
         ctx.flush();
+
+        if (data instanceof ByteBufHttpData) {
+            ((ByteBufHttpData) data).buf().release();
```

Comment:
I just tried it and it didn't work. 

- retainedSlice has refCnt 1, increases buf (parent) refCnt by 1 to total of 2
- buf.release() reduces parent refCnt to 1
- write() reduces retainedSlice refCnt to 0, it gets deallocated, it releases parent, reducing its refCnt from 1 to 0. The parent is deallocated.
- Next call to buf.retainedSlice() doesn't work since buf is already at refCnt 0 and gone.

I'm glad this is what I was expecting :) I rebased and running `./gradlew -Pleak :thrift:test` didn't show anything weird.